docs/ocis/adr/0030-user-generated-client-keys.md

```diff
@@ -0,0 +1,93 @@
+---
+title: "30. User generated client keys"
+date: 2024-02-22T15:30:00+01:00
+weight: 30
+geekdocRepo: https://github.com/owncloud/ocis
+geekdocEditPath: edit/master/docs/ocis/adr
+geekdocFilePath: 0030-user-generated-client-keys.md
+---
+
+* Status: draft
+* Deciders: [@butonic](https://github.com/butonic), [@fschade](https://github.com/fschade), [@kulmann](https://github.com/kulmann), [@dragotin](https://github.com/dragotin), [@rhafer](https://github.com/rhafer)
+* Date: 2024-02-22
+
+## Context and Problem Statement
+
+Currently, the oCIS proxy is used to authenticate users based on an OpenID Connect access token or using basic auth.
+The current implementation is limited to a single identity provider, and we currently have no way to generate credentials for cli tools that do not yet support OpenID Connect
+
+The coming up calendar and contacts service requires this because most CalDAV and CardDAV clients require basic or digest authentication.
```

Comment:
```suggestion
The coming up calendar and contacts service (ccs) requires this because most CalDAV and CardDAV clients require basic or digest authentication.
```